src/groups/mqb/mqba/mqba_domainresolver.cpp

```diff
@@ -85,8 +86,9 @@ void DomainResolver::updateTimestamps()
     d_timestampsValidUntil = now + k_DIR_CHECK_TTL;
 }
 
-bool DomainResolver::cacheLookup(mqbconfm::DomainResolver* out,
-                                 const bslstl::StringRef&  domainName)
+bool DomainResolver::cacheLookup(bsl::string*             resolvedDomainName,
+                                 bsl::string*             clusterName,
+                                 const bslstl::StringRef& domainName)
 {
     // executed by the thread that holds the 'd_mutex'
```

Comment:
```suggestion
    // executed by the thread that holds the 'd_mutex'
    
    // PRECONDITIONS
    BSLS_ASSERT_SAFE(resolvedDomainName);
    BSLS_ASSERT_SAFE(clusterName);
```

Reply:
Added precondition checks and documentation too.